gnucash/gnome-utils/gnc-autoclear.cpp

```diff
@@ -103,6 +105,15 @@ normalize_num (gnc_numeric amount, int denom)
     return normalized.num;
 };
 
+static gint
+count_unique (SplitVec nc_vector, int commodity_scu)
+{
+    std::unordered_set<gint64, Hasher, EqualFn> buckets {};
```

Comment:
This consumes any benefit gained from `sack.reserve` because now the unordered set is allocating chunks for insertion.  See [this SO](https://stackoverflow.com/a/28100858) for a more performant algo to count unique members.

Reply:
Isn't that similar to https://github.com/Gnucash/gnucash/blob/22f83c8697b2945345306fb633e63d0f4c8eb9dc/bindings/guile/utilities.scm#L182 - it requires we define a split sorter which sorts splits in order of inceasing amounts. Sounds like a lot of work for sorting `nc_vector` which is expected to be no more than a couple dozen elements.

Reply:
Not a sorter, just a less-than lambda: 
```
sort(nc_vector.begin(), nc_vector.end(),
       [](Split* s1, Split *s2)->bool { return gnc_numeric_compare(s1->amount, s2->amount) < 0; });
```

But I'll concede that you're right that cost of the ~24 allocations for stuffing the set isn't comparable to the 2<sup>24</sup> insertions into `sack`.